clients/client-py/test/test_client.py

```diff
@@ -310,646 +334,626 @@ def raise_for_status(self):
+def test_success_fifth_try_connection_errors(client, apiPath):
+    with mock.patch.object(utils, 'makeSingleHttpRequest') as p:
+        expected = {'test': 'works'}
+        sideEffect = [
+            requests.exceptions.RequestException,
+            requests.exceptions.RequestException,
+            requests.exceptions.RequestException,
+            requests.exceptions.RequestException,
+            ObjWithDotJson(200, expected)
+        ]
+        p.side_effect = sideEffect
+        expectedCalls = [mock.call('GET', apiPath, None, mock.ANY)
+                            for x in range(client.options['maxRetries'])]
+
+        v = client._makeHttpRequest('GET', 'test', None)
+        p.assert_has_calls(expectedCalls)
+        assert expected == v
+
+def test_failure_status_code(client, apiPath):
+    with mock.patch.object(utils, 'makeSingleHttpRequest') as p:
+        p.return_value = ObjWithDotJson(500, None)
+        expectedCalls = [mock.call('GET',apiPath, None, mock.ANY)
+                            for x in range(client.options['maxRetries'])]
+        with pytest.raises(exc.TaskclusterRestFailure):
+            client._makeHttpRequest('GET', 'test', None)
+        p.assert_has_calls(expectedCalls)
+
+def test_failure_connection_errors(client, apiPath):
+    with mock.patch.object(utils, 'makeSingleHttpRequest') as p:
+        p.side_effect = requests.exceptions.RequestException
+        expectedCalls = [mock.call('GET', apiPath, None, mock.ANY)
+                            for x in range(client.options['maxRetries'])]
+        with pytest.raises(exc.TaskclusterConnectionError):
+            client._makeHttpRequest('GET', 'test', None)
+        p.assert_has_calls(expectedCalls)
+    
+
+
+def test_change_default_doesnt_change_previous_instances(client):
+    prevMaxRetries = subject._defaultConfig['maxRetries']
+    with mock.patch.dict(subject._defaultConfig, {'maxRetries': prevMaxRetries + 1}):
+        assert client.options['maxRetries'] == prevMaxRetries
+
+def test_credentials_which_cannot_be_encoded_in_unicode_work():
+    badCredentials = {
+        'accessToken': u"\U0001F4A9",
+        'clientId': u"\U0001F4A9",
+    }
+    with pytest.raises(exc.TaskclusterAuthFailure):
+        subject.Auth({
+            'rootUrl': base.REAL_ROOT_URL,
+            'credentials': badCredentials,
+        })
 
-class TestMakeApiCall(ClientTest):
-    """ This class covers both the _makeApiCall function logic as well as the
-    logic involved in setting up the api member functions since these are very
-    related things"""
 
-    def setUp(self):
-        ClientTest.setUp(self)
-        patcher = mock.patch.object(self.client, 'NEVER_CALL_ME')
-        never_call = patcher.start()
-        never_call.side_effect = AssertionError
-        self.addCleanup(never_call.stop)
+""" This class covers both the _makeApiCall function logic as well as the
+logic involved in setting up the api member functions since these are very
+related things"""    
```

Comment:
I think this comment no longer applies here. We should probably remove it.

Reply:
Sorry, makes sense. I'll remove it too.

Reply:
This is still here?